File: daily/2022/1689_Partitioning_Into_Minimum_Number_Of_Deci-Binary_Numbers.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    int minPartitions(string n) {
        int M = 0;
        for (auto i : n) M = max(M, i-'0');
        return M;
        // return *max_element(begin(n), end(n)) - '0';
    }
};

int main(){
    // string n = "32";
    // string n = "82734";
    string n = "27346209830709182346";
    Solution S;
    cout << S.minPartitions(n) << endl;
}

File: daily_python/1689_Partitioning_Into_Minimum_Number_Of_Deci_Binary_Numbers.py
import os
import sys
from typing import List

class Solution:
    def minPartitions(self, n: str) -> int:
        return max([ord(c) - ord('0') for c in list(n)])
    
def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            n = lines[0].split('=')[1].strip()[1:-2]
            dataset.append(n)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for n in dataset:
        results.append(solution.minPartitions(n))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')